libcob/fileio.c

```diff
@@ -4150,6 +4170,30 @@ indexed_file_delete (cob_file *f, const char *filename)
 #endif
 }
 
+static int
+indexed_key_compare (const unsigned char *k1, const unsigned char *k2,
+		     size_t sz, const unsigned char *col)
+{
+	if (col == NULL) {
+		return memcmp (k1, k2, sz);
+	} else {
+		return cob_cmps (k1, k2, sz, col);
+	}
+}
+
+static int
+bdb_bt_compare(DB *db, const DBT *k1, const DBT *k2)
+{
+	const unsigned char *col = (unsigned char *)DBT_GET_APP_DATA(k1);
+	if (unlikely (col == NULL)) {
+		cob_runtime_error ("bdb_bt_compare was set but no collating sequence was stored in DBT");
+	}
+	if (unlikely (k1->size != k2->size)) {
```

Comment:
for ease of later migrating to trunk: please drop `unlikely`, as this was removed; instead add LCOV exclusion comments around that, to not decrease coverage because of those checks